### PR TITLE
Document new iterator capabilities

### PR DESCRIPTION
**Pull request by trasher**
Update docs for few things already in 9.3/bf branchs plus:
- https://github.com/glpi-project/glpi/pull/4412
- https://github.com/glpi-project/glpi/pull/4420

### CODE REVIEW

File: source/devapi/database/dbiterator.rst [hunk 1]
```diff
@@ -315,3 +315,85 @@ You can use some aggregation SQL functions on fields: ``COUNT``, ``SUM``, ``AVG`
...
+         'subfield' => 'subvalue'
+      ]
+   ], 'NOT IN');
+   $DB->request(['FROM' => 'glpi_computers', 'WHERE' => ['field' => $sub_query]]);
+   // => SELECT * FROM `glpi_computers` WHERE `field` NOT IN (SELECT `id` FROM `subtable` WHERE `subfield` = 'subvalue')
+
+What if iterator does not provide what I'm looking for?
++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Even if we do our best to get as many things as possible implemented in hte iterator, there are several things that are missing... Consider for example you want to use the SQL `NOW()` function, or want to use a value based on another field: there is no native way to achieve that.
```
Comment (cedric-anne) [inline source/devapi/database/dbiterator.rst]: `hte` -> `the`

File: source/devapi/database/dbupdate.rst [hunk 2]
```diff
@@ -63,9 +52,20 @@ You can update rows in the database using the `update() method <https://forge.gl
 
 An `updateOrDie() method <https://forge.glpi-project.org/apidoc/class-DBmysql.html#_updateOrDie>`_ is also provided.
 
-.. note::
+.. versionadded:: 9.3.1
+
+When issuing an `UPDATE` query, you can use an `ORDER` and/or a `LIMIT` clause alog with the where (which remains **mandatory**). In order to achieve tat, use an indexed array witht appropriate keys:
```
Comment (cedric-anne) [inline source/devapi/database/dbupdate.rst]: `alog` -> `along`
`tat` -> `that`
`witht` -> `with`

File: source/devapi/database/dbupdate.rst [hunk 3]
```diff
@@ -63,9 +52,20 @@ You can update rows in the database using the `update() method <https://forge.gl
...
 
-   The ``update()`` method does not currently support using another field in set. You will therefore have to run queries like ``UPDATE glpi_my_table` SET `ranking` = ranking+1`` using the legacy way.
+.. code-block:: php
+
+   <?php
+   $DB->update(
+      'my_table', [
```
Comment (cedric-anne) [inline source/devapi/database/dbupdate.rst]: Seems that fields to update are missing.

**Comment by trasher** [comment]
Requested changes has been done ;)

**Pull request merged**